lib/runtime-core/src/backend.rs

```diff
@@ -70,6 +70,82 @@ impl std::str::FromStr for Backend {
     }
 }
 
+#[derive(Copy, Clone, Debug)]
+pub enum Architecture {
```

Comment:
At the risk of sounding silly, why do we need this? We ordinarily use cargo CFG to detect what this wasmer build is targeting, so why a runtime value to track it?

Reply:
In case in the future we compile on one architecture and run on another architecture.